File: programmers/Lv_2/HwangJunghoon/H-Index.py
def solution(citations):
    citations.sort(reverse=True)
    answer=0
    for i in range(citations[0], -1, -1):
        cnt=0
        for j in citations:
            if j>=i:
                cnt+=1
        if cnt>=i:
            for k in citations:
                if k!=0:
                    answer=1
                    break
            if answer==0:
                cnt=0
            return min(cnt,i)

File: programmers/Lv_2/JeongJunKwon/H-index.java
import java.util.*;

class Solution {
    public int solution(int[] citations) {
        int index = 0;
        int max = 0;
        Arrays.sort(citations);
        
        for(int i = 0; i < citations.length;) {
            if(index > citations[i]) {
                i++;
                continue;
            }
            
            answer = citations.length - i;
            
            if(answer >= index && i <= index)
                max = index;
            index++;
        }
        return max;
    }
}